nilearn/datasets/struct.py

```diff
@@ -471,55 +479,62 @@ def fetch_surf_fsaverage(mesh='fsaverage5', data_dir=None):
     -------
     data : sklearn.datasets.base.Bunch
         Dictionary-like object, the interest attributes are :
+         - 'area_left': Gifti file, left hemisphere area data
+         - 'area_right': Gifti file, right hemisphere area data
+         - 'curv_left': Gifti file, left hemisphere curvature data
+         - 'curv_right': Gifti file, right hemisphere curvature data
          - 'pial_left': Gifti file, left hemisphere pial surface mesh
          - 'pial_right': Gifti file, right hemisphere pial surface mesh
          - 'infl_left': Gifti file, left hemisphere inflated pial surface mesh
          - 'infl_right': Gifti file, right hemisphere inflated pial
                          surface mesh
+         - 'sphere_left': Gifti file, left hemisphere sphere surface mesh
+         - 'sphere_right': Gifti file, right hemisphere sphere surface mesh
          - 'sulc_left': Gifti file, left hemisphere sulcal depth data
          - 'sulc_right': Gifti file, right hemisphere sulcal depth data
+         - 'thick_left': Gifti file, left hemisphere cortical thickness data
+         - 'thick_right': Gifti file, right hemisphere cortical thickness data
+         - 'white_left': Gifti file, left hemisphere white surface mesh
+         - 'white_right': Gifti file, right hemisphere white surface mesh
 
     References
     ----------
     .. [1] Fischl et al, (1999). High-resolution intersubject averaging and a
        coordinate system for the cortical surface. Hum Brain Mapp 8, 272-284.
 
     """
-    meshes = {'fsaverage5': _fetch_surf_fsaverage5,
-              'fsaverage5_sphere': _fetch_surf_fsaverage5_sphere,
-              'fsaverage': _fetch_surf_fsaverage}
-    if mesh not in meshes:
+    available_meshes = [
+        "fsaverage3", "fsaverage4", "fsaverage5", "fsaverage5_sphere",
+        "fsaverage6", "fsaverage7", "fsaverage",
+    ]
+
+    # Call a dataset loader depending on the value of mesh
+    if (
+        mesh == "fsaverage3"
+        or mesh == "fsaverage4"
+        or mesh == "fsaverage6"
+        or mesh == "fsaverage7"
+        or mesh == "fsaverage"
+    ):
+        # rename mesh to "fsaverage" to download it once
+        # regardless of whether mesh equals "fsaverage" or "fsaverage7"
+        if mesh == "fsaverage7":
+            mesh = "fsaverage"
+
+        return _fetch_surf_fsaverage(mesh, data_dir=data_dir)
+    elif mesh == "fsaverage5":
+        return _fetch_surf_fsaverage5()
+    elif mesh == "fsaverage5_sphere":
+        return _fetch_surf_fsaverage5_sphere(data_dir=data_dir)
```

Comment:
If we deprecate this option, we should add a warning here and not rely on `_fetch_surf_fsaverage5_sphere` (meaning implement the new way).

Reply:
you can use the `sklearn.utils.deprecated` decorator on `_fetch_surf_fsaverage5_sphere` to issue a warning saying to use `fetch_surf_fsaverage("fsaverage5")` instead of `fetch_surf_fsaverage("fsaverage5_sphere")`